source/common/upstream/cluster_manager_impl.cc

```diff
@@ -2399,6 +2399,20 @@ Http::ConnectionPool::InstancePtr ProdClusterManagerFactory::allocateConnPool(
   }
   if (protocols.size() == 1 && protocols[0] == Http::Protocol::Http2 &&
       context_.runtime().snapshot().featureEnabled("upstream.use_http2", 100)) {
+    if (host->cluster().clusterType().has_value() &&
+        host->cluster().clusterType()->name() == "envoy.clusters.reverse_connection") {
+      ENVOY_LOG_MISC(debug, "allocating reverse connection pool");
+      auto* factory = Config::Utility::getFactoryByName<Http::Http2::ReverseConnPoolFactory>(
+          "envoy.http.reverse_conn.default");
+      if (factory) {
+        return factory->allocateConnPool(
+            dispatcher, context_.api().randomGenerator(), server_.singletonManager(), host,
+            priority, options, transport_socket_options, state, origin, alternate_protocols_cache);
+      } else
+        ENVOY_LOG_MISC(error,
```

Comment:
Okay, if it cannot find the factory, should we return here? Since the cluster type is already reverse_connection.

Reply:
Yeah, I think that would be much cleaner